### PR TITLE
Add from_file and write_to_file to SpatialData

### PR DESCRIPTION
**Pull request by vhirtham**
## Changes

- add meshio as new dependency
- add `from_file` and `write_to_file` to `SpatialData`

## Checks

- [x] updated CHANGELOG.md
- [x] updated tests
- [x] updated doc/
- [x] ~~update example/tutorial notebooks~~

### CODE REVIEW

File: weldx/geometry.py [hunk 1]
```diff
@@ -2328,6 +2329,26 @@ def __post_init__(self):
             if not self.triangles.ndim == 2:
                 raise ValueError("SpatialData triangulation must be a 2d array")
 
+    @staticmethod
+    def from_file(file_name: str) -> "SpatialData":
```
Comment (CagtayFabry) [inline weldx/geometry.py:2333]: `pathlib.Path` also seems to be supported by meshio so we might add that to the type hints

Comment (vhirtham) [reply weldx/geometry.py:2333]: yes. I am currently experimenting and writing the test ;)

**Review by marscher — APPROVED** [review]
LGTM!

File: weldx/geometry.py (hunk 1 above)
Comment (marscher) [reply weldx/geometry.py]: It looks like the library is leaking file handles. But using a sane OS this ain't a problem :p

**Comment by marscher** [comment]
> @vhirtham I stumbled upon [this](https://github.com/nschloe/meshio/blob/main/meshio/_files.py) nice utility in `meshio` that might come in handy for other file operations later, just dropping it here :)

This is actually meant to be used when using their readers/writers to avoid leaking file handles!

**Review by CagtayFabry — APPROVED** [review]

**Pull request merged**